plugins/flowdock/app/assets/javascripts/flowdock_plugin.js

```diff
@@ -0,0 +1,112 @@
+samson.factory('Mentionbox', ['$rootScope', 'Flowdock', function ($rootScope, $flowdock) {
+  var self = this;
+  self.users = $flowdock.users();
+
+  self.filteredData = function (query) {
+    return _.filter(self.users, function (item) {
+      return item.name.toLowerCase().indexOf(query.toLowerCase()) > -1;
+    });
+  };
+
+  self.markupData = function (callback) {
+    $(self.mentionsId).mentionsInput('val', function(text) {
+      text = self.reformatMessage(text);
+      callback(text);
+    });
+  };
+
+  self.reformatMessage = function(text) {
+    return text.replace(/@\[([^\\]+?)\]\(([^)]+?)\)/g, "@$1");
+  };
+
+  $rootScope.$on('flowdock_users', function () {
+    self.draw();
+  });
+
+  self.draw = function() {
+    $(self.mentionsId).mentionsInput({
+      elastic: false,
+      defaultValue: self.defaultMessage,
+      onDataRequest:function (mode, query, callback) {
+        callback.call(this, self.filteredData(query));
+      }
+    });
+  };
+
+  self.init = function(id, defaultMessage) {
+    self.mentionsId = id;
+    self.defaultMessage = defaultMessage;
+    self.draw();
+    return {
+      message: self.markupData
+    }
+  };
+
+  return self;
+}]);
+
+samson.factory('Flowdock', ['$rootScope','$http', function ($rootScope, $http) {
+  var self = this;
+
+  self.users = function () {
+    var users = [];
+    var promise = $http.get('/flowdock/users');
+
+    promise.success(function (data) {
+      users.push.apply(users, data.users);
+      $rootScope.$emit('flowdock_users', users);
+    });
+
+    promise.error(function () {
+      console.log('Could not fetch the flowdock users!');
+    });
+
+    return users;
+  };
+
+  self.buddyRequest = function (deploy, message) {
+    return $http.post('/flowdock/notify', { deploy_id: deploy, message: message });
+  };
+
+  return {
+    users: self.users,
+    buddyRequest: self.buddyRequest
+  }
+}]);
+
+samson.controller('BuddyNotificationsCtrl', ['$scope','$rootScope', 'Flowdock', 'Mentionbox',
+  function($scope, $rootScope, flowdock, mentionsBox) {
```

Comment:
Its not a typo. With angular (without ngannotate-rails) using dependency injection I can name whatever I want to that variable, as long as I reference it correctly from the dependency array. From what I am understanding we started using  ngannotate-rails and now I just need to do:
``` 
samson.controller('BuddyNotificationsCtrl', function($scope, $rootScope, flowdock, mentionsBox) {
```
Is it?

Not sure I really like this magic.